be2-scala/src/test/scala/ch/epfl/pop/decentralized/ConnectionMediatorSuite.scala

```diff
@@ -0,0 +1,79 @@
+package ch.epfl.pop.decentralized
+
+import akka.actor.{ActorRef, ActorSystem}
+import akka.testkit.{TestKit, TestProbe}
+import ch.epfl.pop.model.network.method.Heartbeat
+import ch.epfl.pop.pubsub.ClientActor.ClientActorMessage
+import ch.epfl.pop.pubsub.MessageRegistry
+import org.scalatest.BeforeAndAfterAll
+import org.scalatest.funsuite.{AnyFunSuiteLike => FunSuiteLike}
+import org.scalatest.matchers.should.Matchers
+
+class ConnectionMediatorSuite extends TestKit(ActorSystem("ConnectionMediatorSuiteActorSystem")) with FunSuiteLike with Matchers with BeforeAndAfterAll {
```

Comment:
You are not testing the `ConnectTo` Event message, is there a specific reason for that or was it forgotten?

Reply:
To be perfectly honest I wasn't sure to see how to properly test that in a unit test. Surely one could open websocket server in the unit test and have ConnectionMediator try connecting to them. But it looks a bit painful compared to doing it manually or using karate.